PyPDF2/_page.py

```diff
@@ -296,7 +298,7 @@ def __init__(
 
         DictionaryObject.__init__(self)
         self.pdf: Optional[PdfReader] = pdf
-        self.indirect_ref = indirect_ref
+        self.indirect_ref = indirect_ref  # type: ignore[assignment]
```

Comment:
This is a change that is not related to cloning, right? It was just wrong in the tests before.

As this PR is rather big and might take a while to review, I would like to move this to a separate PR ("Miscellaneous fixes" or similar)

Reply:
code reviewed:  a PdfObject indirect_ref can be set to None, to force referencing the object during cloning.